api/envoy/config/common/matcher/v3/matcher.proto

```diff
@@ -226,3 +229,29 @@ message HttpGenericBodyMatch {
   // List of patterns to match.
   repeated GenericTextMatch patterns = 2 [(validate.rules).repeated = {min_items: 1}];
 }
+
+// Matches a specific IP address against a set of possibly overlapping subnets.
+// TODO(kuat) Move to xds repository.
+message IPMatcher {
+  // Specifies an optional list of IP address ranges and a match action.
+  message IPRangeMatcher {
+    // A non-empty set of CIDR ranges.
+    repeated core.v3.CidrRange ranges = 1 [(validate.rules).repeated = {min_items: 1}];
```

Comment:
FYI xds repository lacks `CidrRange` proto so adding this proto there requires that proto as well.

Reply:
It would be fine to copy the `CidrRange` proto to the cncf/xds repo as well.  Eventually, we want everything to move there anyway.

Reply:
Ack, I'll open a separate request in xds repo. We should use the official protos once we are out of experimental.